integrationtest/backend-elasticsearch/src/test/java/org/hibernate/search/v6poc/integrationtest/backend/elasticsearch/search/ElasticsearchSearchSortIT.java

```diff
@@ -0,0 +1,119 @@
+import org.hibernate.search.v6poc.backend.index.spi.ChangesetIndexWorker;
+import org.hibernate.search.v6poc.backend.index.spi.IndexManager;
+import org.hibernate.search.v6poc.backend.index.spi.IndexSearchTarget;
+import org.hibernate.search.v6poc.backend.spatial.GeoPoint;
+import org.hibernate.search.v6poc.backend.spatial.ImmutableGeoPoint;
+import org.hibernate.search.v6poc.engine.spi.SessionContext;
+import org.hibernate.search.v6poc.integrationtest.backend.tck.util.rule.SearchSetupHelper;
+import org.hibernate.search.v6poc.search.DocumentReference;
+import org.hibernate.search.v6poc.search.SearchQuery;
+import org.hibernate.search.v6poc.search.dsl.sort.SearchSortContainerContext;
+import org.hibernate.search.v6poc.util.impl.integrationtest.common.assertion.DocumentReferencesSearchResultAssert;
+import org.hibernate.search.v6poc.util.impl.integrationtest.common.stub.StubSessionContext;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class ElasticsearchSearchSortIT {
+
+	private static final String FIRST_ID = "1";
+	private static final String SECOND_ID = "2";
+	private static final String THIRD_ID = "3";
+	private static final String EMPTY_ID = "empty";
+
+	@Rule
+	public SearchSetupHelper setupHelper = new SearchSetupHelper();
+
+	private IndexAccessors indexAccessors;
+	private IndexManager<?> indexManager;
+	private String indexName;
+	private SessionContext sessionContext = new StubSessionContext();
+
+	@Before
+	public void setup() {
+		setupHelper.withDefaultConfiguration()
+				.withIndex(
+						"MappedType", "IndexName",
+						ctx -> this.indexAccessors = new IndexAccessors( ctx.getSchemaElement() ),
+						(indexManager, indexName) -> {
+							this.indexManager = indexManager;
+							this.indexName = indexName;
+						}
+				)
+				.setup();
+
+		initData();
+	}
+
+	private SearchQuery<DocumentReference> simpleQuery(Consumer<? super SearchSortContainerContext<?>> sortContributor) {
+		IndexSearchTarget searchTarget = indexManager.createSearchTarget().build();
+		return searchTarget.query( sessionContext )
+				.asReferences()
+				.predicate().matchAll().end()
+				.sort( sortContributor )
+				.build();
+	}
+
+	@Test
+	public void byDistanceDesc() {
+		SearchQuery<DocumentReference> query = simpleQuery( b -> b.byDistance( "geoPoint", new ImmutableGeoPoint( 45.757864, 4.834496 ) ).desc() );
+
+		DocumentReferencesSearchResultAssert.assertThat( query )
+				.hasReferencesHitsExactOrder( indexName, EMPTY_ID, SECOND_ID, THIRD_ID, FIRST_ID );
+
+		query = simpleQuery( b -> b.byDistance( "geoPoint", 45.757864, 4.834496 ).desc() );
+
+		DocumentReferencesSearchResultAssert.assertThat( query )
+				.hasReferencesHitsExactOrder( indexName, EMPTY_ID, SECOND_ID, THIRD_ID, FIRST_ID );
+	}
+
+	private void initData() {
+		ChangesetIndexWorker<? extends DocumentElement> worker = indexManager.createWorker( sessionContext );
+		// Important: do not index the documents in the expected order after sorts
```

Comment:
Well, apparently "first/second/third" is not the expected order after sorts here... Maybe something is wrong?

Reply:
Hu?

Reply:
Ah OK :). Fixed then.